include/mbedtls/aes.h

```diff
@@ -57,68 +60,68 @@ extern "C" {
 #endif
 
 /**
- * \brief          AES context structure
- *
- * \note           buf is able to hold 32 extra bytes, which can be used:
- *                 - for alignment purposes if VIA padlock is used, and/or
- *                 - to simplify key expansion in the 256-bit case by
- *                 generating an extra round key
+ * The AES context-type definition. The AES context is passed to the APIs called.
+ * \note           This buffer can hold 32 extra Bytes, which can be used for one of the following purposes:
+ *                 <ul><li>Alignment if VIA padlock is used.</li>
```

Comment:
This comment was in your source - I did not change the content. Should I remove the entire note?

Reply:
Extra compared to the size of an expanded 256-bit key, which is the "natural" size you would expect for this buffer if there weren't any tricks: 14 rounds means 15 round-keys, each worth 4 `uint32_t`, ie a total of 60 `uin32_t`. So `uint32_t buf[68]` is 32 extra bytes compared to that.

Now there are two reasons why a larger buffer is convenient: (1) for AESNI-assisted key schedule, we actually compute as if we needed 16 round key in order to make things more symmetric, so we would need 16 extra bytes for that, and (2) to satisfy padlock alignment constraints.

Now when I look at the definition of `MBEDTLS_PADLOCK_ALIGN16()` if looks to me like `ctx->rk` will point at most 16 bytes after the beginning of `ctx->buf` so that we would only need 16 extra bytes. So that would mean `uint32_t buf[64]` should be enough. Why it was `68` in the first place I honestly don't know - perhaps to accommodate some quirk of the padlock engine (eg it might like 16 extra bytes at the end just like our AESNI code does)? It was already that way (except even less documented) when I added support for AES-NI, and in doubt (and the absence of automated tests on Via Padlock) I didn't dare reduce the size of the buffer.

Reply:
@RosZad Since we're in doubt, please leave the note as it is. We may want to take it out of the typeset documentation and turn it into a more detailed non-Doxygen comment, but we'll do that at our own place.

Reply:
leaving in